### ISSUE TITLE
 AttributeError: 'dict' object has no attribute '_jws'

### ISSUE THREAD

**Issue by furqan-asghar** [GehirnInc/python-jwt#15]
### AttributeError: 'dict' object has no attribute '_jws'
got this error : 

>  File "E:\github\MyGits\TDA\rest_server_sql.py", line 113, in login
    token = jwt.JWT.encode({'public_id' : user.public_id, 'exp' : datetime.datetime.utcnow() + datetime.timedelta(minutes=30)}, app.config['SECRET_KEY'])
  File "E:\github\MyGits\TDA\venv_tda\lib\site-packages\jwt\jwt.py", line 44, in encode
    return self._jws.encode(message, key, alg, optional_headers)
AttributeError: 'dict' object has no attribute '_jws'

while generating token: 
`token = jwt.JWT.encode({'public_id' : user.public_id, 'exp' : datetime.datetime.utcnow() + datetime.timedelta(minutes=30)}, 
                              app.config['SECRET_KEY'])`

**Comment by peterwoo339** [comment]
can anyone shed a light? I've got the same issue..

**Comment by yosida95** [comment]
You should first instantiate the `jwt.JWT` class, then call the `encode` instance method of the instance.

```python3

import jwt

instance = jwt.JWT()
jwt.encode(...)
```

There is also a complete encoding/decoding code example on the README. Refer https://github.com/GehirnInc/python-jwt#examples.

**Comment by nwoow** [comment]
> You should first instantiate the `jwt.JWT` class, then call the `encode` **instance method** of the instance.
> 
> ```python
> import jwt
> 
> instance = jwt.JWT()
> instance.encode(...)
> ```
> 
> There is also a complete encoding/decoding code example on the README. Refer https://github.com/GehirnInc/python-jwt#examples.

I have installed this using:
`pip install jwt`

And I am using a code:

`    
import jwt
instance = jwt.JWT()
 encoded_jwt = instance.encode({'some': 'payload'}, 'secret', algorithm='HS256')`

But I am getting error:

`'function' object has no attribute 'JWT'`

**Comment by yosida95** [comment]
@nwoow Please paste full Python traceback

**Comment by nwoow** [comment]
> @nwoow Please paste full Python traceback

I am using django and I have installed it andv I am trying to generate a token using code above and the response I am getting:
I am getting this error:
```

    instance = jwt.JWT()
AttributeError: 'function' object has no attribute 'JWT'
```

**Comment by yosida95** [comment]
I think that there is a function named `jwt` (`def jwt(...)` or `jwt = ...`) in your script which imports the jwt library because if the jwt imported correctly, jwt is a module instance, not a function.

**Comment by nwoow** [comment]
@yosida95 I have changed my function name to something different actually in views.py controller name was `def jwt` so changed it to the other name and I am getting this error:

```
  instance = jwt.JWT()
AttributeError: module 'jwt' has no attribute 'JWT'
```

**Comment by yosida95** [comment]
@nwoow could you provide the output of `pip freeze`? I think you've installed both jwt and PyJWT.  Due to known name clash (#9) issues, both libraries cannot co-exist. Please also see this issue comment https://github.com/GehirnInc/python-jwt/issues/11#issuecomment-658526304.

**Comment by nwoow** [comment]
@yosida95 yes I am using getstream client and they are using PyJWT both PyJWT and your jwt both are importing jwt how come. Do you have any solution

**Comment by yosida95** [comment]
Unfortunately, no. As of now both libraries use the same `jwt` module namespace and Python's module system cannot resolve `import jwt` deterministically.

**Comment by nwoow** [comment]
@yosida95 anyway I fixe the problem by uninstalling using :

`pip uninstall jwt==1.0.0`

And then I uninstalled pyJWT using:

`pip uninstall PyJWT`

And then install it again:

`pip install PyJWT`

And my code:

    encoded = jwt.encode({'some': 'payload'}, 'secret', algorithm='HS256')

Is working
